### PR TITLE
Aggregate logs

### LINKED ISSUE #2049: Aggregate logs across controllers and services
The current implementation has a combobox to select container within pod:
logs

I would suggest to add second combobox to select pod within a replicaset and keep the selected container

Tracking issue: https://github.com/kubernetes/dashboard/issues/1828

### PR DESCRIPTION
**Pull request by cheld**
Added view logs link to higher controller objects, like replica set. In the logs view the user can select the pod.

Fixes #2049.

(Created 2nd PR after messing up translations)

### CODE REVIEW

**Comment by cheld** [comment]
Injection of logSources is not working with new ui router...

File: src/app/frontend/logs/stateconfig.js [hunk 1]
```diff
@@ -27,16 +27,10 @@ import {stateName} from './state';
  */
 export default function stateConfig($stateProvider) {
   $stateProvider.state(stateName, {
-    url: `${appendDetailParamsToUrl('/log')}/:container`,
-    params: {
-      'container': {
```
Comment (cheld) [inline src/app/frontend/logs/stateconfig.js]: I removed that. Is that ok

File: src/app/frontend/logs/component.js [hunk 2]
```diff
@@ -62,20 +60,29 @@ export class LogsController {
...
+     */
+    this.container;
 
-    /** @export {!backendApi.PodContainerList} - initialized from resolve */
-    this.podContainers;
+    /**
+     * Pods and containers available for selection
+     * @export {!backendApi.LogSources}
+     */
+    this.logSources;
```
Comment (cheld) [inline src/app/frontend/logs/component.js]: This is not injected, but 'this.podLogs' is set. I don't see any difference. Not sure about the new mechanism...

File: src/app/frontend/logs/stateconfig.js (hunk 1 above)
Comment (floreks) [inline src/app/frontend/logs/stateconfig.js]: No you have to rewrite that to match `resourceType` param. Logs page won't work correctly without thtat.

File: src/app/frontend/logs/component.js (hunk 2 above)
Comment (floreks) [inline src/app/frontend/logs/component.js]: `podLogs` is injected through component definition. Remove `podContainers` from there and add `logSources`.

Comment (cheld) [inline src/app/frontend/logs/component.js]: Thanks

File: src/app/frontend/logs/stateconfig.js (hunk 1 above)
Comment (cheld) [inline src/app/frontend/logs/stateconfig.js]: hmmm...container is optional, thus (as far as I understand) default value is null and squash is true. However, resourceType is not optional.

Comment (floreks) [inline src/app/frontend/logs/stateconfig.js]: No. It has changed.

https://ui-router.github.io/guide/ng1/migrate-to-1_0#non-optional-path-parameters

**Comment by cheld** [comment]
PTAL

**Comment by cheld** [comment]
What is the problem with coverage. Any idea?

**Comment by maciaszczykm** [comment]
@cheld Probably some Codecov issue. On our side it seems to be fine: https://travis-ci.org/kubernetes/dashboard/jobs/254430310#L694

**Comment by floreks** [comment]
Master build failed during coverage report creation and there was no coverage to compare against.

_Sent from my Google Nexus 5X using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

File: src/app/backend/handler/apihandler.go [hunk 3]
```diff
@@ -61,6 +57,10 @@ import (
 	errorsK8s "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/kubernetes/pkg/client/unversioned/remotecommand"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
```
Comment (floreks) [inline src/app/backend/handler/apihandler.go]: Was that formatted by your IDE? Try change it to `goimports` style or run `goimports -w src/app/backend` in dashboard root directory. Previous format is enforced by `goimports`. I have to update `gulp format`.

Comment (cheld) [inline src/app/backend/handler/apihandler.go]: @floreks goimports changes about 20 files. Should I change as part of this PR? WDYT?

Comment (floreks) [inline src/app/backend/handler/apihandler.go]: Yes. Formatting files is fine.

**Comment by cheld** [comment]
PTAL

File: src/app/backend/client/manager.go [hunk 4]
```diff
@@ -20,7 +20,6 @@ import (
 	"log"
 	"strings"
 
-	"github.com/emicklei/go-restful"
```
Comment (floreks) [inline src/app/backend/client/manager.go]: In every package `go-restful` import is removed. Backend is not building now.

Comment (cheld) [inline src/app/backend/client/manager.go]: hmm...seems to be a bug if import name has a hyphen (https://github.com/joefitzgerald/go-plus/issues/52)

Comment (floreks) [inline src/app/backend/client/manager.go]: Install and use [this](https://godoc.org/golang.org/x/tools/cmd/goimports) `go get golang.org/x/tools/cmd/goimports`. I have used it and it doesn't delete anything.

**Comment by cheld** [comment]
I would suggest to fix goimport in a separate PR, e.g. adding it to gulp format. WDYT?

**Comment by floreks** [comment]
Revert changes and make CI pass. We can format it later.

**Pull request merged**